lib/cog/command/command_resolver.ex

```diff
@@ -1,54 +1,73 @@
 defmodule Cog.Command.CommandResolver do
-
   alias Cog.Repo
-  alias Cog.Queries.Command
-  alias Cog.Queries.Alias, as: AliasQuery
-  alias Piper.Command.SemanticError
+  alias Cog.Models.Command
+  alias Cog.Models.UserCommandAlias
+  alias Cog.Models.SiteCommandAlias
+  alias Cog.Queries
 
-  def find_bundle(<<":", _::binary>>=name, user) do
-    find_bundle_or_alias(name, user)
-  end
-  def find_bundle(name, user) when is_binary(name) do
-    if String.contains?(name, ":") do
-      :identity
-    else
-      find_bundle_or_alias(name, user)
+  @doc """
+  Resolver function for use in `Piper.Command.ParserOptions.resolver`.
+  """
+  def command_resolver_fn(user) do
+    fn(ns, name) ->
+      case lookup(ns, name, user) do
+        %Command{}=c ->
+          {:command, {c.bundle.name, c.name, c}}
+        %UserCommandAlias{pipeline: pipeline} ->
+          {:pipeline, pipeline}
+        %SiteCommandAlias{pipeline: pipeline} ->
+          {:pipeline, pipeline}
+        :not_found ->
+          :not_found
+        {:ambiguous, names} ->
+          {:ambiguous, names}
+      end
```

Comment:
@lhaskins `command_resolver_fn` returns a function that closes over the passed in user. This function is passed into the parser and is used as a callback to lookup the relevant information; the parser determines the bundle name and command name to be passed in based on the parsed text.